core/network/src/protocol.rs

```diff
@@ -1301,16 +1316,29 @@ impl<B: BlockT, S: NetworkSpecialization<B>, H: ExHashT> Protocol<B, S, H> {
 		who: PeerId,
 		request: message::RemoteReadChildRequest<B::Hash>,
 	) {
+		let keys_str = match request.keys.len() {
```

Comment:
I guess it's not a performance bottleneck, but we could do that lazily, only when logging is enabled. I guess it's better to do it twice when tracing than to do it even if we don't use it at all.